odl/operator/tensor_ops.py

```diff
@@ -337,7 +337,14 @@ def derivative(self, vf):
         inner_vf = vf.copy()
 
         for gi in inner_vf:
-            gi /= vf_pwnorm_fac * gi ** (self.exponent - 2)
+            if self.exponent >= 2:
```

Comment:
Should be `< 2`, no? The code handles the division-by-zero case, which does not correspond to this condition.

Reply:
I understood the problem as being that for any ``exponent``, if the element contains a zero component the code will divide by 0. For ``exponent < 2`` the derivative is actually undefined in this point (and hence we let it divide by zero...), but for ``exponent >= 2`` the derivative is actually zero in that direction. Thus the safe-guarding is only in the case of ``exponent >= 2``.

Have I misunderstood the derivative?

Reply:
Ah, I was under the assumption that you solve the `0 ** (exponent - 2)` issue. Now the distinction makes sense.

Anyway, I would still suggest a different implementation for exponent < 2, namely
```python
gi *= gi ** (2 - self.exponent) / vf_pwnorm_fac
```
It also seems to me that we should have `np.abs(gi) ** (2 - self.exponent)` and `np.abs(gi) ** (self.exponent - 2)`, respectively. Compare with the formula.